installer/scripts/service_control

```diff
@@ -20,29 +23,75 @@ VAR_DIR=/var/opt/microsoft/omsagent
 ETC_DIR=/etc/opt/microsoft/omsagent
 BIN_DIR=/opt/microsoft/omsagent/bin
 
-WORKSPACE_REGEX='^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'
+REGEX_UUID='[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}'
+WORKSPACE_REGEX="^$REGEX_UUID$"
+VALIDATION_SUPPLEMENT_MESSAGE=
+VALIDATION_SEVERITY=WARNING
 
 # Space seperated list of non oms workspaces
 NON_OMS_WS="scom"
 
-setup_variables()
-{
+# Begin Lower Level Utilities
+
+validate_file_has_line_pattern() {
+	local _fSpec=$1
+	local _pattA="$2"
+	local _boolOnly=$3
+	cat $_fSpec | grep -Eq "^$_pattA$"
+	if [ $? -eq 0 ]; then
+		return 0
+	fi
+	if [ -z "$_boolOnly" ]; then
+		echo "$VALIDATION_SEVERITY:  $_fSpec missing required pattern '$_pattA'."
+        echo $VALIDATION_SUPPLEMENT_MESSAGE
+	fi
+	exit 1
+}
+
+# End Lower Level Utilities
+
+# Begin FailCount Singleton Interface
+
+fcso_any_failures() {
+    return $FCSO_FailCount
+}
+
+fcso_increment() {
+    if [ $# -ne 3 ]; then
+        echo "FATAL:  Programmer error; 3 arguments required for fcso_increment!"
+        echo "$# arguments seen."
+        exit 99
+    fi
+    local _actStr=$1
+    local _wId=$2
+    local _sCode=$3
+
+    if [ "$_sCode" -eq 0 ]; then
```

Comment:
Both types should be string: `"$_sCode" -eq "0"`

Reply:
Yep.  Thanks.